radix-engine/src/blueprints/consensus_manager/consensus_manager.rs

```diff
@@ -1414,6 +1446,10 @@ impl ConsensusManagerBlueprint {
             )?;
         }
 
+        // For any reason, if a validator isn't included in the `validator_infos` but has accumulated
+        // proposer rewards, we reset the counter as the rewards has been distributed to other validators.
+        validator_rewards.proposer_rewards.clear();
```

Comment:
(optional) I would somehow feel safer if we captured the "not applicable validators" (e.g. from line 1289 `// Excluded due to slashing ?` I guess?) and explicitly removed them from `validator_rewards` before the `for` loop (at 1416). We could even assert that the map is empty at the method end.